internal/xds/translator/route.go

```diff
@@ -224,7 +224,7 @@ func buildXdsStringMatcher(irMatch *ir.StringMatch) *matcherv3.StringMatcher {
 
 func buildXdsRouteAction(backendWeights *ir.BackendWeights, settings []*ir.DestinationSetting) *routev3.RouteAction {
 	// only use weighted cluster when there are invalid weights
-	if hasFiltersInSettings(settings) || backendWeights.Invalid != 0 {
+	if hasFiltersInSettings(settings) || backendWeights.Invalid != 0 || hasMultiBackendsAndZoneAwareRouting(settings) {
```

Comment:
similar reuse of https://github.com/envoyproxy/gateway/pull/5299/files#r1980217608